src/frontend/packages/store/src/effects/recursive-entity-delete.effect.ts

```diff
@@ -87,10 +95,12 @@ export class RecursiveDeleteEffect {
       const tree = this.getTree(action, state);
       const actions = new Array<Action>().concat(...Object.keys(tree).map<Action[]>(key => {
         const keyActions = [];
-        if (this.deleteSuccessApiActionGenerators[key]) {
-          keyActions.push(this.deleteSuccessApiActionGenerators[key](action.guid, action.endpointGuid));
+        const deleteSuccessApiActionGenerators = action.entityConfig.definition.recursiveDelete ?
```

Comment:
This all feels like we're subverting the pipelines. If this is to become a generic part of the codebase and is expected to be used by extensions then it should follow the same action dispatch flow (even better, use the pipes and handlers used to build the api pipeline).

How would an extensions that uses a custom pipeline leverage the recursive delete?

APISuccessOrFailedAction is considered an internal action and, I would image, we would not make it available for extensions to use in the long run. Also, why do we only dispatch that action for applications and not all of the other entity types.